File: MPP/MPP Proiect/Java/Persistence/src/main/java/persistence/repository/dbrepositories/ParticipantRepository.java
package persistence.repository.dbrepositories;

import model.Participant;
import org.apache.logging.log4j.Logger;
import persistence.repository.interfaces.IParticipantRepository;
import persistence.repository.utils.JDBCUtils;
import org.apache.logging.log4j.LogManager;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Properties;

public class ParticipantRepository implements IParticipantRepository {

    protected JDBCUtils jdbcUtils;

    private static final Logger logger = LogManager.getLogger();

    public ParticipantRepository(Properties properties){
        this.jdbcUtils = new JDBCUtils(properties);
        logger.info("Initializing ParticipantRepository with properties: {} ", properties);
    }

    @Override
    public void add(Participant entity) {
        logger.traceEntry("Saving task {}", entity);
        Connection con = jdbcUtils.getConnection();
        try (PreparedStatement preStnt = con
                .prepareStatement("insert into Participants(name, totalPoints) values (?, ?)")) {
            preStnt.setString(1, entity.getName());
            preStnt.setInt(2, entity.getTotalPoints());
            int result = preStnt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit("Successfully saved participant");
    }

    @Override
    public void update(Participant entity) {
        logger.traceEntry("Saving task {}", entity);
        Connection con = jdbcUtils.getConnection();
        try (PreparedStatement preStnt = con
                .prepareStatement("update Participants set totalPoints = ? where id_p = ?")) {
            preStnt.setInt(1, entity.getTotalPoints());
            preStnt.setInt(2, entity.getId());
            int result = preStnt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit("Successfully saved participant");
    }

    @Override
    public void remove(Participant entity) {
        this.removeById(entity.getId());
    }


    @Override
    public void removeById(int id) {
        logger.traceEntry("Saving task {}", id);
        Connection con = jdbcUtils.getConnection();
        try (PreparedStatement preStnt = con
                .prepareStatement("delete from Participants where id_p = ?")) {
            preStnt.setInt(1, id);
            int result = preStnt.executeUpdate();
            logger.trace("Deleted {} instances", result);
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit("Successfully deleted participant");
    }

    @Override
    public ArrayList<Participant> getAll() {
        logger.traceEntry();
        Connection con = jdbcUtils.getConnection();
        ArrayList<Participant> participants = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from Participants")){
            try(ResultSet result = preStmt.executeQuery()){
                while(result.next()){
                    int id = result.getInt("id_p");
                    String name = result.getString("name");
                    int totalPoints = result.getInt("totalPoints");
                    Participant participant = new Participant(name, totalPoints);
                    participant.setId(id);
                    participants.add(participant);
                }
            }
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return participants;
    }

    @Override
    public Participant getById(Integer id) {
        logger.traceEntry();
        Connection con = jdbcUtils.getConnection();
        Participant participant = null;
        try(PreparedStatement preStmt = con.prepareStatement("select * from Participants where id_p = " + id)){
            try(ResultSet result = preStmt.executeQuery()){
                if(result.next()){
                    int id_p = result.getInt("id_p");
                    String name = result.getString("name");
                    int totalPoints = result.getInt("totalPoints");
                    participant = new Participant(name, totalPoints);
                    participant.setId(id_p);
                }
            }
        } catch (SQLException ex) {
            logger.error((ex));
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return participant;
    }

}

File: MPP/MPP Proiect/CSharp/persistance/DBRepositories/ParticipantRepository.cs
using System.Collections.Generic;
using System.Data.SQLite;
using log4net;
using model;
using persistance.Interfaces;
using persistance.Utils;

namespace persistance.DBRepositories{
    public class ParticipantRepository: IParticipantRepository{
        private static readonly ILog log =
            LogManager.GetLogger("ParticipantRepository");

        private IDictionary<string, string> _props;

        public ParticipantRepository(IDictionary<string, string> props){
            this._props = props;
            log.Info("Creating ParticipantRepository");
        }

        public void Add(Participant entity){
            log.Info("ParticipantRepository Add method");
            log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            using(var command = new SQLiteCommand(
                      "insert into Participants values (@id, @name, @totalPoints)",
                      connection as SQLiteConnection)){
                command.Parameters.AddWithValue("@id", null);
                command.Parameters.AddWithValue("@name", entity.Name);
                command.Parameters.AddWithValue("@totalPoints", entity.TotalPoints);
                command.ExecuteNonQuery();
                log.InfoFormat("Succesfully added participant {0}", entity);
            }
        }


        public void Remove(Participant entity){
            log.Info("ParticipantRepository RemoveByID method");
            RemoveById(entity.id);
        }

        public void RemoveById(int id){
            log.Info("ParticipantRepository RemoveByID method");
            log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            using(var command = new SQLiteCommand(
                      "delete from Participants where id_p = @id",
                      connection as SQLiteConnection)){
                command.Parameters.AddWithValue("@id", id);
                command.ExecuteNonQuery();
                log.InfoFormat("Succesfully removed participant with id {0}", id);
            }
        }

        public List<Participant> GetAll(){
            var list = new List<Participant>();
            log.Info("ParticipantRepository GetAll method");
            log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            using(var cmd = new SQLiteCommand("select * from Participants",
                      connection as SQLiteConnection)){
                using(var reader = cmd.ExecuteReader()){
                    while(reader.Read()){
                        var id = reader.GetInt32(0);
                        var name = reader.GetString(1);
                        var totalPoints = reader.GetInt32(2);
                        var participant = new Participant(name, totalPoints);
                        participant.id = id;
                        list.Add(participant);
                    }
                }
            }

            log.Info("Successfully created participant's list");
            return list;
        }

        public Participant GetById(int id){
            Participant participant = null;
            log.Info("ParticipantRepository GetById method");
            log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            var cmd = new SQLiteCommand(
                "select * from Participants where id_p = " + id,
                connection as SQLiteConnection);
            using(var reader = cmd.ExecuteReader()){
                if(reader.Read()){
                    var name = reader.GetString(1);
                    var totalPoints = reader.GetInt32(2);
                    participant = new Participant(name, totalPoints);
                    participant.id = id;
                }
            }

            log.Info("Successfully got participant");
            return participant;
        }

        public void Update(Participant participant){
            log.Info("ParticipantRepository Update method");
            log.Info("Get database connection");
            var connection = DbUtils.GetConnection(_props);
            using(var command = new SQLiteCommand(
                      "update Participants set name = @name, totalPoints = @totalPoints where id_p = @id",
                      connection as SQLiteConnection)){
                command.Parameters.AddWithValue("@id", participant.id);
                command.Parameters.AddWithValue("@name", participant.Name);
                command.Parameters.AddWithValue("@totalPoints",
                    participant.TotalPoints);
                command.ExecuteNonQuery();
                log.InfoFormat("Succesfully updated participant {0}", participant);
            }
        }
    }
}